vir/src/make.rs

```diff
@@ -602,7 +670,7 @@ impl<'tcx> VirCtxt<'tcx> {
 
     pub fn mk_exhale_stmt<'vir, Curr, Next>(
         &'vir self,
-        expr: ExprGen<'vir, Curr, Next>,
+        expr: ExprGenBool<'vir, Curr, Next>,
```

Comment:
I guess this isn't something that's part of the type system yet, but the `exhale` expression is a more general "impure" type (which includes e.g. `acc` expressions). Maybe add a TODO to the type system file if we ever want to revisit this.